Fix empty broker ID on LoanBrokerSet

gregtatcam opened the pull request:
## High Level Overview of Change

Fix [RIPD-3920](https://ripplelabs.atlassian.net/browse/RIPD-3920)

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Performance (increase or change in throughput and/or latency)
- [ ] Tests (you added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation update
- [ ] Chore (no impact to binary, e.g. `.gitignore`, formatting, dropping support for older tooling)
- [ ] Release

## Test Plan

Extend LoanBroker unit-tests.

gregtatcam committed:
Fix empty/zero brokerID, extend unit-tests.

ximinez reviewed (changes requested):
So it looks like these tests demonstrate that all the `LoanBroker` tx types except `LoanBrokerSet` already handle the zero correctly, and then you fixed `LoanBrokerSet`.

Nice.

There's one more transaction that I'm concerned about: `LoanSet`. Could you add a test for that, and fix it if needed? (It would probably be fine to even put it in `LoanBroker_test` and reuse the helper.)

gregtatcam committed:
Fix empty/zero brokerID in LoanSet, extend unit-tests.

gregtatcam commented:
> So it looks like these tests demonstrate that all the `LoanBroker` tx types except `LoanBrokerSet` already handle the zero correctly, and then you fixed `LoanBrokerSet`.
> 
> Nice.
> 
> There's one more transaction that I'm concerned about: `LoanSet`. Could you add a test for that, and fix it if needed? (It would probably be fine to even put it in `LoanBroker_test` and reuse the helper.)

I fixed LoanSet and added the test to Loan_test to be consistent with how invalid tests are structured. Besides LoanSet is slightly  different from LoanBroker because of the CounterpartySignature.

ximinez commented on src/xrpld/app/tx/detail/LoanSet.cpp:
```diff
@@ -119,6 +119,10 @@ LoanSet::preflight(PreflightContext const& ctx)
             return *ret;
     }
 
+    if (auto const brokerID = ctx.tx[~sfLoanBrokerID];
+        brokerID && *brokerID == beast::zero)
+        return temINVALID;
```
sfLoanBrokerID is required, so this doesn't have to be optional, but it's fine as-is.

gregtatcam replied:
Ah, sorry. I must have looked at a different tx.

ximinez committed:
Merge branch 'ximinez/lending-XLS-66' into gregtatcam/bugs/ripd-3920

ximinez committed:
Merge branch 'ximinez/lending-XLS-66' into gregtatcam/bugs/ripd-3920
